h2o-algos/src/main/java/hex/tree/PlattScalingHelper.java

```diff
@@ -0,0 +1,124 @@
+package hex.tree;
+
+import hex.Model;
+import hex.ModelBuilder;
+import hex.ModelCategory;
+import hex.glm.GLM;
+import hex.glm.GLMModel;
+import water.*;
+import water.fvec.Frame;
+import water.fvec.Vec;
+
+import static hex.ModelCategory.Binomial;
+
+public class PlattScalingHelper {
+    
+    public interface ModelBuilderWithCalibration<M extends Model<M , P, O>, P extends Model.Parameters, O extends Model.Output> {
+        void warn(String field, String msg);
+        void error(String field, String msg);
+        int nclasses();
+        Frame init_adaptFrameToTrain(Frame f, String label, String param, boolean expensive);
+        Frame getCalibrationFrame();
+        void setCalibrationFrame(Frame f);
+        void setCalibrationOutput(GLM calibrationOutput, M model);
+    }
+
+    public interface ParamsWithCalibration {
+        boolean isCVModel();
+        String weightsColumn();
+        String responseColumn();
+        
+        Frame getCalibrationFrame();
+        boolean calibrateModel();
+    }
+
+    public interface OutputWithCalibration {
+        ModelCategory getModelCategory();
+        GLMModel calibrationModel();
+    }
+    
+    public static void initCalibration(ModelBuilderWithCalibration builder, ParamsWithCalibration parms, boolean expensive) {
+        // Calibration
+        Frame cf = parms.getCalibrationFrame();  // User-given calibration set
+        if (cf != null) {
+            if (! parms.calibrateModel())
+                builder.warn("_calibration_frame", "Calibration frame was specified but calibration was not requested.");
+            Frame adaptedCf = builder.init_adaptFrameToTrain(cf, "Calibration Frame", "_calibration_frame", expensive);
+            builder.setCalibrationFrame(adaptedCf);
+        }
+        if (parms.calibrateModel()) {
+            if (builder.nclasses() != 2)
+                builder.error("_calibrate_model", "Model calibration is only currently supported for binomial models.");
+            if (cf == null)
+                builder.error("_calibrate_model", "Calibration frame was not specified.");
+        }
+    }
+    
+    public static <M extends Model<M , P, O>, P extends Model.Parameters, O extends Model.Output> void buildCalibrationModel(
+        boolean finalScoring, ModelBuilderWithCalibration<M, P, O> builder, 
+        ParamsWithCalibration parms, Job job, M model
+    ) {
+        // Model Calibration (only for the final model, not CV models)
+        if (finalScoring && parms.calibrateModel() && (!parms.isCVModel())) {
```

Comment:
maybe calibrateModel could also consider whether this is a cvModel - then on we could move the logic on the caller side

   if (finalScoring && params.calibrateModel())

IMHO better than to have the logic here

Reply:
done